### PR TITLE
add unit tests for precision, recall and f1_score metrics

### PR DESCRIPTION
**Pull request by dcpagotto**
Hey! This PR adds comprehensive unit tests for the precision, recall and f1_score metrics that were missing test coverage.

Previously these 3 metrics had zero tests while mean_average_precision was the only one with tests. This was mentioned in issue #2108 which asked for increasing test coverage to ~90%.

What I tested:
- Instance creation with default and custom parameters
- Reset functionality to clear stored data 
- Perfect matches, no overlaps, empty cases
- Single class and multiple class scenarios  
- Different IoU thresholds (0.5 vs 0.75)
- All 3 averaging methods (macro, micro, weighted)
- Confidence ranking and edge cases
- List inputs and error handling

All test values were calculated manually to make sure the math is correct. Added comments explaining the calculations so its easy to verify.

The tests follow the same style as existing tests in the repo and use pytest with parametrize for testing different configurations.

This should help get closer to the 90% coverage target mentioned in the issue.

### CODE REVIEW

File: test/metrics/test_f1_score.py [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+    predictions = Detections(
+        xyxy=np.array([[10, 10, 50, 50]], dtype=np.float32),
+        confidence=np.array([0.9]),
+        class_id=np.array([0]),
+    )
+
+    targets = Detections(
+        xyxy=np.empty((0, 4), dtype=np.float32),
+        class_id=np.empty((0,), dtype=int),
+    )
```
Comment (Borda) [inline test/metrics/test_f1_score.py]: seems you are using the same, so lets wrap then to a class and use fixture to init these two inputs to umprove redability

**Review by Borda — APPROVED** [review]
